single_kernel_mongo/core/abstract_upgrades.py

```diff
@@ -336,11 +337,16 @@ def run_post_upgrade_checks(self, finished_whole_cluster: bool = False) -> None:
         raise NotImplementedError()
 
     # BEGIN: Helpers
+    @mongodb_only
     def move_primary_to_last_upgrade_unit(self) -> None:
         """Moves the primary to last unit that gets upgraded (the unit with the lowest id).
 
         Raises FailedToMovePrimaryError
         """
+        # This should only ever run on mongos
+        if self.dependent.name == CharmKind.MONGOS:
+            logger.error("move_primary_to_last_upgrade_unit called on mongos charm. Please fix.")
```

Comment:
is this log correct/necessary? I wouldn't be sure what to do if I was a user in this case

Reply:
Actually, I have implemented some decorators for that and I just forgot to remove this check here…
Thanks, I'm cleaning this up.